core/packet.c

```diff
@@ -109,31 +110,46 @@ static coap_status_t handle_request(lwm2m_context_t * contextP,
     coap_status_t result = NOT_FOUND_4_04;
 
     uriP = lwm2m_decode_uri(message->uri_path);
-    if (uriP == NULL) return BAD_REQUEST_4_00;
-
-    switch(uriP->flag & LWM2M_URI_MASK_TYPE)
-    {
+    if (uriP == NULL) {
```

Comment:
Do you mean in the very special case of bootstrapping or in a more generic case? If it's in a generic case, I would prefer create a new branch for that and not mix a related fix in the di_bootstrap branch (so this one could be merged without having to rebase too often onto master...)